packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -242,26 +242,28 @@ abstract class ImageStreamCompleter extends Diagnosticable {
     }
     if (_currentError != null && onError != null) {
       try {
-          onError(_currentError.exception, _currentError.stack);
-        } catch (exception, stack) {
-          FlutterError.reportError(
-            new FlutterErrorDetails(
-              exception: exception,
-              library: 'image resource service',
-              context: 'by a synchronously-called image error listener',
-              stack: stack,
-            ),
-          );
-        }
+        onError(_currentError.exception, _currentError.stack);
+      } catch (exception, stack) {
+        FlutterError.reportError(
+          new FlutterErrorDetails(
+            exception: exception,
+            library: 'image resource service',
+            context: 'by a synchronously-called image error listener',
+            stack: stack,
+          ),
+        );
+      }
     }
   }
 
   /// Stop listening for new concrete [ImageInfo] objects and errors from
   /// its associated [ImageErrorListener].
   void removeListener(ImageListener listener) {
-    _listeners.removeWhere((_ImageListenerPair listenerPair) {
-      return listenerPair.listener == listener;
-    });
+    _listeners.remove(
+      _listeners.firstWhere((_ImageListenerPair listenerPair) {
+        return listenerPair.listener == listener;
+      })
+    );
```

Comment:
i suspect this is expensive relative to just walking the list and removing the first match by index.